src/Components/ShareControl.jsx

```diff
@@ -0,0 +1,153 @@
+import React, {createRef, useEffect, useState} from 'react'
+import Checkbox from '@mui/material/Checkbox'
+import FormControlLabel from '@mui/material/FormControlLabel'
+import FormGroup from '@mui/material/FormGroup'
+import TextField from '@mui/material/TextField'
+import ToggleButton from '@mui/material/ToggleButton'
+import {makeStyles} from '@mui/styles'
+import {ControlButton} from './Buttons'
+import Dialog from './Dialog'
+import {
+  addCameraUrlParams,
+  hasValidUrlParams as urlHasCameraCoords,
+  removeCameraUrlParams,
+} from './CameraControl'
+import {assertDefinedBoolean} from '../utils/assert'
+import CopyIcon from '../assets/2D_Icons/Copy.svg'
+import ShareIcon from '../assets/2D_Icons/Share.svg'
+import CameraIcon from '../assets/2D_Icons/Camera.svg'
+
+
+/**
+ * This button hosts the ShareDialog component and toggles it open and
+ * closed.
+ * @param {Object} viewer ifc viewer
+ * @return {Object} The button react component, with a hosted
+ *   ShareDialog component
+ */
+export default function ShareControl({viewer}) {
+  const [isDialogDisplayed, setIsDialogDisplayed] = useState(false)
+  return (
+    <ControlButton
+      title='Share this model'
+      icon={<ShareIcon/>}
+      isDialogDisplayed={isDialogDisplayed}
+      setIsDialogDisplayed={setIsDialogDisplayed}
+      dialog={
+        <ShareDialog
+          viewer={viewer}
+          setIsDialogDisplayed={setIsDialogDisplayed}/>
+      }/>
+  )
+}
+
+
+/**
+ * The ShareDialog component lets the user control what state is
+ * included in the shared URL and assists in copying the URL to
+ * clipboard.
+ * @param {Object} viewer IFC viewer
+ * @param {function} setIsDialogDisplayed
+ * @return {Component} The react component
+ */
+function ShareDialog({viewer, setIsDialogDisplayed}) {
+  const [isLinkCopied, setIsLinkCopied] = useState(false)
+  const [isCameraInUrl, setIsCameraInUrl] = useState(assertDefinedBoolean(urlHasCameraCoords()))
+  const urlTextFieldRef = createRef()
+  const classes = useStyles()
+
+  useEffect(() => {
+    if (viewer) {
+      if (isCameraInUrl) {
+        addCameraUrlParams(viewer)
+      } else {
+        removeCameraUrlParams()
+      }
+    }
+  }, [viewer, isCameraInUrl])
+
+  const closeDialog = () => {
+    setIsDialogDisplayed(false)
+    setIsLinkCopied(false)
+  }
+
+  const onCopy = (event) => {
+    setIsLinkCopied(true)
+    navigator.clipboard.writeText(location)
+    urlTextFieldRef.current.select()
+  }
+
+  const toggleCameraIncluded = () => {
+    if (isCameraInUrl) {
+      setIsCameraInUrl(false)
+    } else {
+      setIsCameraInUrl(true)
+    }
+    if (isLinkCopied) {
+      setIsLinkCopied(false)
+    }
+  }
+
+  const CameraButton = () => {
+    return (
+      <ToggleButton value='cameraInclude' selected={isCameraInUrl}>
+        <CameraIcon/>
```

Comment:
app className = classes.icon 
width:'30px'
height:'30px'
to fix the ellipse icon.
<img width="499" alt="image" src="https://user-images.githubusercontent.com/3433606/156637404-55870b57-82aa-4eed-994b-51234ba3c239.png">

Reply:
Adding note here as todo.. clicking on links in portfolio appends full URL to existing URL..

Reply:
Not sure what was up with it.. had to remodel it and remove the style altogether.

NB: width and height are controlled for all of the new buttons in Buttons#useStyles